### DISCUSSION TITLE
Project HÆLTH Blog #4

### DISCUSSION THREAD

**Discussion by j1tbk** [ldcdorn/haelth#48] [General]
# Project HÆLTH 🌱
Our fourth blog post covers the topics we worked on and goals we achieved this week.

#  Classes
We created the first classes for our project which will be used for future developement and provide an outline for the developpers on our team. Our design focus on simplicity and functionality resulting in a seemless and pleasant user experience. 
Each subsystem as presented by our UML-Diagrams has it's own class which can track scores, progress and personal information.
For further visualisation we created a Class-Diagram in [Draw.io](url) as follows:

![Classes drawio](https://github.com/user-attachments/assets/4556cc35-cf7d-43cf-b1d9-63f86db40413)

Furthermore, the classes have been added to the project as such:

```
class DB {
    val path: String
    var version: String

    fun DB() {
        // TODO
    }

    fun checkIntegrity() {
        // TODO
    }
}

class Fitness {
    var score: Int = 0
    var exercisePlan = List()

    fun Fitness() {

    }

    fun getAcitivtiy() {
        // TODO
    }

}

class GoalManager {
    var goals = List()

    fun GoalManager() {

    }

    fun add() {
        // TODO
    }

    fun remove() {
        // TODO
    }
}

class Mindfulness {
    var score: Int
    var exercises = List()

    fun Mindfulness() {
        // TODO
    }
}

class Nutrition {
    var score: Int
    var mealHistory = List()

    fun Nutrition() {
        // TODO
    }
}

class UI {
    val version : String

    public UI() {
        //TODO
    }

    fun update() {
        // TODO
    }

    fun refresh() {
        // TODO
    }
}	

class User {
    var name: String
    var age: Int
    val sex

    fun User() {
        // TODO
    }
}
```

**Comment by themsgcentre** [comment]
Vielen Dank für den Einblick in eure Fortschritte und die Struktur, die ihr in eurem Blog vermittelt. Es ist spannend zu sehen, wie ihr mit Klassendiagrammen arbeitet und euren Entwicklungsprozess Schritt für Schritt festhaltet. Die wöchentliche Zusammenfassung hilft, den Verlauf und eure Gedankengänge gut nachzuvollziehen – besonders für die, die von außen einen Blick darauf werfen möchten.

Das Klassendiagramm zeigt, wie durchdacht eure Planung ist und illustriert bemerkenswert, wie die verschiedenen Komponenten zusammenhängen. Die Dokumentation eurer Vorgehensweise ist hilfreich und inspirierend. Das Einzige, was uns aufgefallen ist, ist, dass bei den Kompositionen noch die Anzahl/Kardinalitäten fehlen.

Wir freuen uns auf weitere Updates und sind gespannt, wie sich das Projekt weiterentwickelt!

Viele Grüße,
Projekt TDC (HM)
